tron/eventbus.py

```diff
@@ -0,0 +1,162 @@
+import logging
+import os
+import pickle
+import time
+from collections import deque
+
+from twisted.internet import reactor
+
+log = logging.getLogger(__name__)
+
+
+def make_eventbus(log_dir):
+    return EventBus(log_dir)
+
+
+def consume_dequeue(queue, func):
+    queue_length = len(queue)
+    for _ in range(queue_length):
+        func(queue.popleft())
+
+
+class EventBus:
+    def __init__(self, log_dir):
+        self.must_shutdown = False
+        self.event_log = {}
+        self.event_subscribers = {}
+        self.publish_queue = deque()
+        self.subscribe_queue = deque()
+        self.log_dir = log_dir
+        self.log_current = os.path.join(self.log_dir, "current")
+        self.log_updates = 0
+        self.log_last_save = 0
+        self.log_save_interval = 60
+        self.log_save_updates = 10
+
+    def start(self):
+        log.info("starting")
+        self.setup_log_dir()
+        self.sync_load_log()
+        reactor.callLater(0, self.sync_loop)
+
+    def setup_log_dir(self):
```

Comment:
im not convinced this is the right place for this - can you just pass a file path into the constructor and expect the caller to deal with this?